### PR TITLE
Change public to external on join and exit

### PR DESCRIPTION
**Pull request by talbaneth**
# Description

Collateral Type (e.g. `ETH-A`):

COB Team Name or Author(s):

GemJoin Adapter:

Is this GemJoin adapter new? :

If so, why is a new one needed? :

# Contribution Checklist

- [ ] Collateral token contract has been added in `src/tokens/` as `TOKEN.sol`, if not already present
- [ ] If needed, a new `GemJoin` has been added in `src/` as `join-X.sol`
- [ ] A new test (`testGemJoinX_TOKEN()`) has been added to `src/join.t.sol` and follows previous tests as examples
- [ ] If needed, other tests have been added to verify functionality of the new `GemJoin`:
    - [ ] `testFailGemJoinXJoinWad()`
    - [ ] `testFailGemJoinXExitWad()`
    - [ ] `testFailGemJoinXJoin()`
    - [ ] `testFailGemJoinXExit()`
    - [ ] `testFailJoinAfterCageGemJoinX()`
- [ ] README has been updated to reflect a new token and `GemJoin` addition

# Checklist

- [ ] Confirm that token contract matches the verified source on etherscan mainnet
- [ ] If a new `GemJoin` is added, ensure the above description is accurate
- [ ] If a new `GemJoin` is added, ensure a new `GemJoin` has been added with the correct name
- [ ] Ensure README is updated
- [ ] Manually review the new test(s)
- [ ] Manually run dapp tests in `src/join.t.sol` to confirm they all pass

### CODE REVIEW

**Review by hexonaut — APPROVED** [review]
LGTM 👍

File: src/join-managed.sol [hunk 1]
```diff
@@ -90,11 +90,15 @@ contract ManagedGemJoin {
         emit Join(usr, amt);
     }
 
-    function exit(address usr, uint256 amt) external {
-        exit(usr, msg.sender, amt);
+    function exit(address usr, uint256 amt) external auth {
```
[the remark this answers is not in the document]
Comment (brianmcmichael) [reply src/join-managed.sol:93]: re: auth function, this was calling the other `public` function, but instead the logic for both was moved to `internal` and the auth modifier was added to the external signatures.

[the remark this answers is not in the document]
Comment (brianmcmichael) [reply src/join-managed.sol:93]: I see the issue with the params. We should make the switch but I'd still vote for keeping the function for flexibility in the future. It's entirely possible that we can have several authed contracts pointing to one managed gem join.

[the remark this answers is not in the document]
Comment (talbaneth) [reply src/join-managed.sol]: As discussed offline I've pushed a change to alter the params to align with other gemJoins (in functionality). We are leaning towards removing this new exit() but will further discuss.

File: src/join-managed.sol [hunk 2]
```diff
@@ -90,11 +90,15 @@ contract ManagedGemJoin {
         emit Join(usr, amt);
     }
 
-    function exit(address usr, uint256 amt) external {
-        exit(usr, msg.sender, amt);
+    function exit(address dst, uint256 amt) external auth {
```
[the remark this answers is not in the document]
Comment (hexonaut) [reply src/join-managed.sol:93]: Ok as long as it has been discussed. Whatever you guys decide I'm fine with.

File: src/join-managed.sol [hunk 3]
```diff
@@ -90,11 +90,7 @@ contract ManagedGemJoin {
         emit Join(usr, amt);
     }
 
-    function exit(address usr, uint256 amt) external {
-        exit(usr, msg.sender, amt);
-    }
-
-    function exit(address usr, address dst, uint256 amt) public auth {
+    function exit(address usr, address dst, uint256 amt) external auth {
```
[the remark this answers is not in the document]
Comment (gbalabasquer) [reply src/join-managed.sol:93]: Hmmm I think I'd prefer `exit(address urn, address usr, uint256 amt)` or rename all the other joins to `exit(address dst, uint256 amt)`

**Pull request merged**